pkg/provider/azure_privatelinkservice.go

```diff
@@ -162,16 +162,19 @@ func (az *Cloud) reconcilePrivateLinkService(
 	} else if !wantPLS {
 		existingPLS, err := az.getPrivateLinkService(fipConfigID, azcache.CacheReadTypeDefault)
 		if err != nil {
-			klog.Errorf("reconcilePrivateLinkService for service(%s): getPrivateLinkService(%s) failed: %v", serviceName, pointer.StringDeref(fipConfigID, ""), err)
-			return err
-		}
-
-		exists := !strings.EqualFold(pointer.StringDeref(existingPLS.ID, ""), consts.PrivateLinkServiceNotExistID)
-		if exists {
-			deleteErr := az.safeDeletePLS(&existingPLS, service)
-			if deleteErr != nil {
-				klog.Errorf("reconcilePrivateLinkService for service(%s): deletePLS for frontEnd(%s) failed: %v", serviceName, pointer.StringDeref(fipConfigID, ""), err)
-				return deleteErr.Error()
+			if !retry.HasStatusForbiddenOrIgnoredError(err) {
```

Comment:
It seems it will not return not found error if failed to get the pls by fipConfigID?

Reply:
Yes, it only leaves a warning. I think it is ok because it is in `if !wantPLS` condition.

Reply:
If the condition will not met, why bother to add so many lines of code for a scenario that will never happen?